frontend/app-preview/src/components/AppBarConfig/AppPreviewBarConfig.tsx

```diff
@@ -102,18 +101,16 @@ export const appPreviewButtonActions = (
   app: string,
   instanceId: string,
 ): AltinnButtonActionItem[] => {
-  const subUrl = `/${RoutePaths.UIEditor}?layout=`;
+  const packagesRouter = new PackagesRouter({ org, app });
+  const subUrl = `?layout=${window.localStorage.getItem(instanceId)}`;
+
   const action: AltinnButtonActionItem[] = [
     {
       title: 'top_menu.preview_back_to_editing',
-      path: editorPath,
       menuKey: TopBarMenu.Preview,
       buttonVariant: 'secondary',
       headerButtonsClasses: classes.backToEditorBtn,
-      handleClick: () =>
-        (window.location.href = `${editorPath(org, app)}${subUrl}${window.localStorage.getItem(
-          instanceId,
-        )}`),
+      handleClick: () => packagesRouter.navigateToPackage('editorUiEditor', subUrl),
```

Comment:
we should renaming `subUrl` to `queryParams`. 😊 
```suggestion
      handleClick: () => packagesRouter.navigateToPackage('editorUiEditor', queryParams),
```